### PR TITLE
`ProductRequestData`: added `Storefront` for receipt posting

### PR DESCRIPTION
**Pull request by NachoSoto**
Supersedes #1499.
~~Depends on #1502.~~
~~Will need to wait for #1504 to be able to generate correct snapshots.~~
Fixes [CF-434]

[CF-434]: https://revenuecats.atlassian.net/browse/CF-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Tests/UnitTests/Purchasing/ProductRequestDataInitializationTests.swift [hunk 1]
```diff
@@ -7,15 +7,20 @@ import XCTest
 class ProductRequestDataSK1ProductInitializationTests: XCTestCase {
 
     private var product: MockSK1Product!
+    private var storefront: MockStorefront!
 
     private static let productID = "cool_product"
+    private static let storefrontCountryCode = "ESP"
```
Comment (taquitos) [inline Tests/UnitTests/Purchasing/ProductRequestDataInitializationTests.swift:13]: 🇪🇸

**Review by taquitos — APPROVED** [review]
👯‍♂️ teamwork makes the dream work
🎈 🐐

File: Tests/UnitTests/Purchasing/ProductRequestDataInitializationTests.swift (hunk 1 above)
Comment (NachoSoto) [reply Tests/UnitTests/Purchasing/ProductRequestDataInitializationTests.swift:13]: 😇

**Comment by NachoSoto** [comment]
This is updated now, but might need new snapshots for iOS 12 @taquitos

**Comment by taquitos** [comment]
@NachoSoto storefront is showing up for iOS 12 tests now, I'm like 99.9% sure it should only be for iOS 13 and above

**Comment by NachoSoto** [comment]
Is it not only for the ones where we're mocking the storefront?

**Comment by taquitos** [comment]
> Is it not only for the ones where we're mocking the storefront?

I think that's right, storefront should be nil for iOS 12

**Comment by NachoSoto** [comment]
Yeah, I didn't make the mock conditionally ignore it based on the version.

**Comment by NachoSoto** [comment]
If you want to push the iOS 12 snapshots here we can merge it :D

**Comment by taquitos** [comment]
All iOS 12 snapshots are uploaded with the correct json, so we just need to remove the mocked storefront for the following tests:
Screen Shot 2022-04-15 at 8 59 33 AM

**Comment by NachoSoto** [comment]
Oh cool, so you manually deleted the fields?
Sure I can make the mock "smarter"

**Comment by taquitos** [comment]
> Oh cool, so you manually deleted the fields? Sure I can make the mock "smarter"

I generated the files before the mock existed, so the correct data was already in there 😁

**Comment by NachoSoto** [comment]
Do you mind pushing those?

**Comment by taquitos** [comment]
> Do you mind pushing those?

I think you force-pushed over them yesterday, ha. Let me regenerate.

**Comment by taquitos** [comment]
Oh, wait, they are already up there: https://github.com/RevenueCat/purchases-ios/tree/main/Tests/UnitTests/Networking/Backend/__Snapshots__/BackendPostReceiptDataTests

**Comment by NachoSoto** [comment]
Oh shoot sorry!

**Comment by taquitos** [comment]
> Oh shoot sorry!

No no, my bad, I added them to a different pr that got merged, you're good 😄

**Comment by taquitos** [comment]
Backend tests will fail until backend task is completed

**Comment by NachoSoto** [comment]
Will update these today.

**Comment by taquitos** [comment]
> Will update these today.

I **_think_** the only remaining item is that the backend needs an update so the backend integration tests pass again. 

I noticed that the iOS 15 tests were crashing and restarting over and over again- so that's one thing to check into, but I was kinda hoping it was a CI hiccup.

File: Sources/Purchasing/ProductRequestData.swift [hunk 2]
```diff
@@ -59,7 +59,7 @@ extension ProductRequestData: Encodable {
         case productIdentifier = "product_id"
         case paymentMode = "payment_mode"
         case currencyCode = "currency"
-        case storefront
+        case storefront = "store_country"
```
Comment (NachoSoto) [inline Sources/Purchasing/ProductRequestData.swift:62]: 👍🏻

**Review by NachoSoto — COMMENTED** [review]
Awesome

**Pull request merged**